app/models/account.rb

```diff
@@ -120,7 +120,7 @@ def groups
   end
 
   def add_group(owner)
-    with_valid_account_type('user') { set_parent(owner) }
+    with_valid_account_type('user') { add_parent(owner) }
```

Comment:
Yeah, those are some of the intentional "multiple parents", so having a method that is specifically designed for that seems clearer.